### ISSUE TITLE
[BUG] [2.4.0] Install error on Mac "error: use of undeclared identifier 'SQL_GD_OUTPUT_PARAMS'"

### ISSUE THREAD

**Issue by dstotz** [markdirish/node-odbc#225]
**Describe your system**
* `odbc` Package Version: 2.4.0
* ODBC Driver: n/a
* Database Name: n/a
* Database Version: n/a
* Database OS: n/a
* Node.js Version: `16.9.0`, `14.17.1`, `14.16.0`, `12.16.2`
* Node.js OS: Mac OS Catalina 10.15.7

**Describe the bug**
Unable to install version `2.4.0` on Mac. Install fails with error `error: use of undeclared identifier 'SQL_GD_OUTPUT_PARAMS'`

**Expected behavior**
`npm install odbc@2.4.0` should complete successfully

**To Reproduce**
1. `npm init` create new project
2. `npm install odbc@2.4.0`

**Code**
If applicable, add snippets of code to show:

* Any DEBUG information printed to the terminal:

```bash
➜  node-odbc-mac-test npm install

> odbc@2.4.0 install /Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! install response status 404 Not Found on https://github.com/markdirish/node-odbc/releases/download/v2.4.0/odbc-v2.4.0-darwin-x64-napi-v5.tar.gz 
node-pre-gyp WARN Pre-built binaries not installable for odbc@2.4.0 and node@12.16.2 (node-v72 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/markdirish/node-odbc/releases/download/v2.4.0/odbc-v2.4.0-darwin-x64-napi-v5.tar.gz 
  CXX(target) Release/obj.target/odbc/src/odbc.o
../src/odbc.cpp:466:50: error: use of undeclared identifier 'SQL_GD_OUTPUT_PARAMS'
        (bool) (sql_getdata_extensions_bitmask & SQL_GD_OUTPUT_PARAMS);
                                                 ^
1 error generated.
make: *** [Release/obj.target/odbc/src/odbc.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/derek.stotz/.nvm/versions/node/v12.16.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/Users/derek.stotz/.nvm/versions/node/v12.16.2/bin/node" "/Users/derek.stotz/.nvm/versions/node/v12.16.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5/odbc.node" "--module_name=odbc" "--module_path=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5" "--napi_version=5" "--node_abi_napi=napi" "--napi_build_version=5" "--node_napi_label=napi-v5"
gyp ERR! cwd /Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc
gyp ERR! node -v v12.16.2
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/Users/derek.stotz/.nvm/versions/node/v12.16.2/bin/node /Users/derek.stotz/.nvm/versions/node/v12.16.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5/odbc.node --module_name=odbc --module_path=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5 --napi_version=5 --node_abi_napi=napi --napi_build_version=5 --node_napi_label=napi-v5' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp ERR! System Darwin 19.6.0
node-pre-gyp ERR! command "/Users/derek.stotz/.nvm/versions/node/v12.16.2/bin/node" "/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc
node-pre-gyp ERR! node -v v12.16.2
node-pre-gyp ERR! node-pre-gyp -v v1.0.5
node-pre-gyp ERR! not ok 
Failed to execute '/Users/derek.stotz/.nvm/versions/node/v12.16.2/bin/node /Users/derek.stotz/.nvm/versions/node/v12.16.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5/odbc.node --module_name=odbc --module_path=/Users/derek.stotz/Downloads/node-odbc-mac-test/node_modules/odbc/lib/bindings/napi-v5 --napi_version=5 --node_abi_napi=napi --napi_build_version=5 --node_napi_label=napi-v5' (1)
npm WARN odbc-mac-test@1.0.0 No description
npm WARN odbc-mac-test@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! odbc@2.4.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the odbc@2.4.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/derek.stotz/.npm/_logs/2021-09-09T14_47_55_895Z-debug.log
```

**Additional context**

Got the same error on all 4 versions of Node that I tested. This version installs fine on Windows, and version `2.3.6` installs fine on both Windows and Mac.

Mac is an i9, so not an m1 chip issue.

**Comment by kadler** [comment]
Do you have unixODBC installed via brew? ` brew info unixodbc`

**Comment by dstotz** [comment]
@kadler I did have unixodbc `2.3.9` installed and it was working for `odbc@2.3.6` but failing for `odbc@2.4.0` but updated it and it grabbed version 2.3.9_1` and that seems to have fixed the issue!

**Comment by kadler** [comment]
That's weird. AFAIK, 2.3.9_1 only has the fix I made for ARM64 Macs. Not sure why it would affect SQL_GD_OUTPUT_PARAMS. Maybe one of the headers got messed with previously and updating the package caused them to be replaced? In any case, glad things are working now.

**Comment by dstotz** [comment]
@kadler I may have jumped to a conclusion that the `2.3.9_1` upgrade is what actually fixed the issue. I scrolled back up and looked through the output a bit more and it looks like there was also a missing dependency with my `2.3.9` install of `libtool`
![image](https://user-images.githubusercontent.com/9381594/132727214-77111dd6-6601-4e05-9c13-1e30705894e4.png)

It also failed to symlink automatically on the update so I had to do `brew link --overwrite unixodbc` as well. Maybe there was a missing/bad symlink and that just overwriting that fixed it.
